### PR TITLE
docs(microservices): add commit offsets doc for kafka

### PR DESCRIPTION
**Pull request by davidschuette**
This is the associated documentation for PR #9270 of nest.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/docs.nestjs.com/blob/master/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [x] Docs
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: [PR #9270 of nestjs/nest](https://github.com/nestjs/nest/pull/9270)

## What is the new behavior?
Add documentation for usage of `commitOffsets` for ClientKafka.

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

### CODE REVIEW

File: content/microservices/kafka.md [hunk 1]
```diff
@@ -389,3 +389,59 @@ onModuleInit() {
 ```
 
 > info **Hint** Kafka reply topic naming conventions can be customized by extending `ClientKafka` in your own custom provider and overriding the `getResponsePatternName` method.
+
+#### Commit offsets
+
+Comitting offsets is essential when working with kafka. Per default messages will be automatically committed after a specific time. For more information visit [kafkajs docs](https://kafka.js.org/docs/consuming#autocommit). `ClientKafka` offers a way to manually commit offsets that functions like the [native kafkajs implementation](https://kafka.js.org/docs/consuming#manual-committing).
```
Comment (johannesschobel) [inline content/microservices/kafka.md:395]:
```suggestion
Committing offsets is essential when working with Kafka. Per default, messages will be automatically committed after a specific time. For more information visit [KafkaJS docs](https://kafka.js.org/docs/consuming#autocommit). `ClientKafka` offers a way to manually commit offsets that work like the [native KafkaJS implementation](https://kafka.js.org/docs/consuming#manual-committing).
```

File: content/microservices/kafka.md [hunk 2]
```diff
@@ -389,3 +389,59 @@ onModuleInit() {
...
+async handleUserCreated(@Payload() data: IncomingMessage) {
+  // business logic
+
+  await this.client.commitOffsets([ { topic: data.topic, partition: data.partition, offset: data.offset } ])
+}
+@@switch
+
+constructor(@Inject(KAFKA) client: ClientKafka) {}
+
+@EventPattern('user_created')
```
Comment (johannesschobel) [inline content/microservices/kafka.md:412]: Is using `_` (underscore) a good idea? Because it is rarely used in the TypeScript world, i guess..

**Review by johannesschobel — CHANGES_REQUESTED** [review]

**Comment by davidschuette** [comment]
@johannesschobel Thank you for the review. I will update the PR with your suggestions and with some more content when this [PR](https://github.com/nestjs/nest/pull/9293) gets merged.

File: content/microservices/kafka.md [hunk 3]
```diff
@@ -389,3 +389,59 @@ onModuleInit() {
...
+async handleUserCreated(data) {
+  // business logic
+
+  await this.client.commitOffsets([ { topic: data.topic, partition: data.partition, offset: data.offset } ])
+}
+```
+
+To disable auto-committing of messages set `autoCommit: false` in the `run` config.
```
Comment (kamilmysliwiec) [inline content/microservices/kafka.md:420]:
```suggestion
To disable auto-committing of messages set `autoCommit: false` in the `run` configuration, as follows:
```

File: content/microservices/kafka.md [hunk 4]
```diff
@@ -389,3 +389,59 @@ onModuleInit() {
...
+@@filename()
+
+constructor(@Inject(KAFKA) client: ClientKafka) {}
+
+@EventPattern('user_created')
+async handleUserCreated(@Payload() data: IncomingMessage) {
+  // business logic
+
+  await this.client.commitOffsets([ { topic: data.topic, partition: data.partition, offset: data.offset } ])
+}
```
Comment (kamilmysliwiec) [inline content/microservices/kafka.md:407]:
```suggestion
@EventPattern('user.created')
async handleUserCreated(@Payload() data: IncomingMessage, @Ctx() context: KafkaContext) {
  // business logic
  
  const originalMessage = context.getMessage();
  const { topic, partition, offset } = originalMessage;
  await this.client.commitOffsets([{ topic, partition, offset }])
}
```

File: content/microservices/kafka.md [hunk 5]
```diff
@@ -389,3 +389,59 @@ onModuleInit() {
...
+@@switch
+
+constructor(@Inject(KAFKA) client: ClientKafka) {}
+
+@EventPattern('user_created')
+async handleUserCreated(data) {
+  // business logic
+
+  await this.client.commitOffsets([ { topic: data.topic, partition: data.partition, offset: data.offset } ])
+}
```
Comment (kamilmysliwiec) [inline content/microservices/kafka.md:417]:
```suggestion
@Bind(Payload(), Ctx())
@EventPattern('user.created')
async handleUserCreated(data, context) {
  // business logic

  const originalMessage = context.getMessage();
  const { topic, partition, offset } = originalMessage;
  await this.client.commitOffsets([{ topic, partition, offset }])
}
```

**Pull request merged**